### PR TITLE
chore: upgrade group/ff/halo2 and rust version

### PR DESCRIPTION
**Pull request by zhenfeizhang**
### Description

[_PR description_]

### Issue Link

[_link issue here_]

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Contents

- [_item_]

### Rationale

[_design decisions and extended information_]

### How Has This Been Tested?

[_explanation_]

<hr>

## How to fill a PR description 

Please give a concise description of your PR.

The target readers could be future developers, reviewers, and auditors. By reading your description, they should easily understand the changes proposed in this pull request.

MUST: Reference the issue to resolve

### Single responsability

Is RECOMMENDED to create single responsibility commits, but not mandatory.

Anyway, you MUST enumerate the changes in a unitary way, e.g.

```
This PR contains:
- Cleanup of xxxx, yyyy
- Changed xxxx to yyyy in order to bla bla
- Added xxxx function to ...
- Refactored ....
```

### Design choices

RECOMMENDED to:
- What types of design choices did you face?
- What decisions you have made?
- Any valuable information that could help reviewers to think critically

### CODE REVIEW

File: aggregator/src/tests/rlc/gates.rs [hunk 1]
```diff
@@ -76,120 +77,120 @@ impl Circuit<Fr> for ArithTestCircuit {
...
+                // // unit test: is equal
+                // {
+                //     let should_be_false = config.is_equal(&mut region, &f1, &f2, &mut offset)?;
+                //     let zero = config.load_private(&mut region, &Fr::zero(), &mut offset)?;
+                //     region.constrain_equal(should_be_false.cell(), zero.cell())?;
+
+                //     let should_be_true = config.is_equal(&mut region, &f1, &f1, &mut offset)?;
+                //     let one = config.load_private(&mut region, &Fr::one(), &mut offset)?;
+                //     region.constrain_equal(should_be_true.cell(), one.cell())?;
+                // }
```
Comment (lispc) [inline aggregator/src/tests/rlc/gates.rs]: all the commented codes are useless?

Comment (lispc) [reply aggregator/src/tests/rlc/gates.rs]: seems yes

**Comment by lispc** [comment]
testing result


tx circuit failures may be due to disabled SigCircuit?
```
failures:
    evm_circuit::execution::sar::test::test_sar_gadget_with_max_values
    evm_circuit::execution::sar::test::test_sar_gadget_with_negative_a
    evm_circuit::execution::sar::test::test_sar_gadget_with_positive_a
    evm_circuit::execution::sar::test::test_sar_gadget_with_random_values
    tx_circuit::test::tx_circuit_0tx_1max_tx
    tx_circuit::test::tx_circuit_1tx_1max_tx
    tx_circuit::test::tx_circuit_1tx_2max_tx
    tx_circuit::test::tx_circuit_2tx_2max_tx
    tx_circuit::test::tx_circuit_l1_msg_tx
    tx_circuit::test::tx_circuit_to_is_zero
```

**Comment by lispc** [comment]
current codes seems ok for me.  What is the status of sig circuit and ecc circuit? They were disabled due to halo2-libs, but currently halo2-libs are upgraded to ff13, so they can be recovered?

**Comment by lispc** [comment]
another question is why we upgrade rust here? some dep needs it? @zhenfeizhang

**Comment by zhenfeizhang** [comment]
> another question is why we upgrade rust here? some dep needs it? @zhenfeizhang

 `marker::ConstParamTy` requires a newer version of rust

**Comment by zhenfeizhang** [comment]
> current codes seems ok for me. What is the status of sig circuit and ecc circuit? They were disabled due to halo2-libs, but currently halo2-libs are upgraded to ff13, so they can be recovered?



yes

File: aggregator/src/tests/rlc/gates.rs (hunk 1 above)
Comment (zhenfeizhang) [reply aggregator/src/tests/rlc/gates.rs]: those are useful unit tests that were commented out by mistake. they are not uncommented

**Comment by lispc** [comment]
@zhenfeizhang there may be a lot of clippy warning not easy to fix due to rust upgrade.  So HHQ Zhang will handle clippy fix later.

**Comment by lispc** [comment]
https://github.com/scroll-tech/zkevm-circuits/pull/1022/commits/a4a54b1a665e8e1ecac9addacfcd4fc2fe0aeaef may not be correct. Can be reverted

**Comment by lispc** [comment]
https://github.com/scroll-tech/zkevm-circuits/actions/runs/7108484917/job/19351858425?pr=1022 CI fails..  more memory usage?   We need to bench speed & memory after these upgrades..

**Comment by lispc** [comment]
\b run testool 1

**Review by lispc — APPROVED** [review]

**Pull request merged**